src/main/java/com/ibm/disni/util/MemBuf.java

```diff
@@ -53,39 +48,6 @@ public final ByteBuffer getBuffer() {
 		return buffer;
 	}
 
-	public MemoryAllocation.MemType getType() {
-		return type;
-	}
-	
-	public String getClassname() {
-		return this.classname;
-	}	
-	
-	public void dump() {
-		dump(0, buffer.capacity());
-	}
-	
-	public void dump(int offset, int length) {
-		MemBuf.dump(buffer, offset, length);
-	}
-	
-	public static void dump(ByteBuffer buffer, int offset, int length) {
-		if (offset + length > buffer.capacity()) {
-			return;
-		}
+	public final int size() { return buffer.capacity(); }
 
-		String outputbuf = "";
-		for (int i = offset; i < offset + length; i++) {
-			byte tmp1 = buffer.get(i);
-			int tmp2 = tmp1 & 0xff;
-			outputbuf += tmp2 + " ";
-		}
-		logger.info(outputbuf);
-	}
-
-	public void free() {
-		if (memAlloc != null){
-			memAlloc.free(this);
-		}
-	}
 }
```

Comment:
Most of this looks good. I agree that we can siplify MemBuf (no need to manage type, etc.). But why removing free()? I think it's nice to have each MemBuf know how it can de-allocate itself, it avoid users to have access to MemoryAllocation or even to know who is the memory allocation manager for this object.